### DISCUSSION TITLE
Admob crashes Android app on build

### DISCUSSION THREAD

**Discussion by waldemirflj** [invertase/react-native-firebase#4759] [Q&A]
Help me I'm trying to implement Firebase / Admob and it doesn't work.

react-native: 0.63.3
@react-native-firebase/app: 10.4.0
@react-native-firebase/admob: 10.4.0

`Configure project :react-native-firebase_admob
:react-native-firebase_admob package.json found at /home/waldemir/www/xxxxx/node_modules/@react-native-firebase/admob/package.json
:react-native-firebase_app package.json found at /home/waldemir/www/xxxxx/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_admob:firebase.bom using default value: 26.2.0
:react-native-firebase_admob:ads.consent using default value: 1.0.6
:react-native-firebase_admob package.json found at /home/waldemir/www/xxxxx/node_modules/@react-native-firebase/admob/package.json
:react-native-firebase_admob:version set from package.json: 10.4.1 (10,4,1 - 10004001)
:react-native-firebase_admob:android.compileSdk using custom value: 29
:react-native-firebase_admob:android.targetSdk using custom value: 29
:react-native-firebase_admob:android.minSdk using custom value: 16
:react-native-firebase_admob:reactNativeAndroidDir /home/waldemir/www/xxxxx/node_modules/react-native/android

> Configure project :react-native-firebase_app
:react-native-firebase_app package.json found at /home/waldemir/www/xxxxx/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_app:firebase.bom using default value: 26.2.0
:react-native-firebase_app:play.play-services-auth using default value: 19.0.0
:react-native-firebase_app package.json found at /home/waldemir/www/xxxxx/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_app:version set from package.json: 10.4.0 (10,4,0 - 10004000)
:react-native-firebase_app:android.compileSdk using custom value: 29
:react-native-firebase_app:android.targetSdk using custom value: 29
:react-native-firebase_app:android.minSdk using custom value: 16
:react-native-firebase_app:reactNativeAndroidDir /home/waldemir/www/xxxxx/node_modules/react-native/android

> Configure project :react-native-firebase_database
:react-native-firebase_database package.json found at /home/waldemir/www/xxxxx/node_modules/@react-native-firebase/database/package.json
:react-native-firebase_app package.json found at /home/waldemir/www/xxxxx/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_database:firebase.bom using default value: 26.2.0
:react-native-firebase_database package.json found at /home/waldemir/www/xxxxx/node_modules/@react-native-firebase/database/package.json
:react-native-firebase_database:version set from package.json: 10.4.1 (10,4,1 - 10004001)
:react-native-firebase_database:android.compileSdk using custom value: 29
:react-native-firebase_database:android.targetSdk using custom value: 29
:react-native-firebase_database:android.minSdk using custom value: 16
:react-native-firebase_database:reactNativeAndroidDir /home/waldemir/www/xxxxx/node_modules/react-native/android

> Configure project :react-native-firebase_messaging
:react-native-firebase_messaging package.json found at /home/waldemir/www/xxxxx/node_modules/@react-native-firebase/messaging/package.json
:react-native-firebase_app package.json found at /home/waldemir/www/xxxxx/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_messaging:firebase.bom using default value: 26.2.0
:react-native-firebase_messaging package.json found at /home/waldemir/www/xxxxx/node_modules/@react-native-firebase/messaging/package.json
:react-native-firebase_messaging:version set from package.json: 10.4.1 (10,4,1 - 10004001)
:react-native-firebase_messaging:android.compileSdk using custom value: 29
:react-native-firebase_messaging:android.targetSdk using custom value: 29
:react-native-firebase_messaging:android.minSdk using custom value: 16
:react-native-firebase_messaging:reactNativeAndroidDir /home/waldemir/www/xxxxx/node_modules/react-native/android

> Task :app:processDebugResources FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.2/userguide/command_line_interface.html#sec:command_line_warnings
151 actionable tasks: 2 executed, 149 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > Android resource linking failed
     /home/waldemir/.gradle/caches/transforms-2/files-2.1/c771f6e95aa7685585546b88818fb45b/play-services-ads-lite-19.6.0/AndroidManifest.xml:27:5-38:15: AAPT: error: unexpected element <queries> found in <manifest>.
         

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 4s

error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup. Run CLI with --verbose flag for more details.
Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
`

**Comment by andersonaddo** [comment]
Have you tried cleaning Gradle and React Native cache?

**Reply by mikehardy** [reply]
That's my thought too. @waldemirflj ---> `npx react-native-clean-project`
Please note our test project builds, as does https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh

**Reply by waldemirflj** [reply]
> Have you tried cleaning Gradle and React Native cache?

Yes. I've been clearing the cache for almost 2 weeks.

**Reply by waldemirflj** [reply]
> That's my thought too. @waldemirflj ---> `npx react-native-clean-project`
> Please note our test project builds, as does https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh

I'll try.
